src/Microsoft.Framework.Configuration/ConfigurationSection.cs

```diff
@@ -38,64 +41,79 @@ public IEnumerable<IConfigurationSource> Sources
             }
         }
 
-        public string Get([NotNull] string key)
+        public string Key
         {
-            string value;
-            return TryGet(key, out value) ? value : null;
+            get
+            {
+                return _key;
+            }
+        }
+
+        public string Value
+        {
+            get
+            {
+                if (_value != null)
```

Comment:
As far as I know the value can change underneath us and we shouldn't cache it locally. Remove `_value`.